scripts/enumerable.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select = []
+    my_each { |item| select << item if yield item }
+    select
+  end
+
+  def my_all?(class_passed = nil)
+    all = true
+    if block_given?
+      my_each { |item| all = false unless yield item }
+    elsif class_passed
+      my_each { |item| all = false unless item.is_a? class_passed }
+    else
+      my_each { |item| all = false unless item }
+    end
+    all
+  end
+
+  def my_any?(class_passed = nil)
+    any = false
+    if block_given?
+      my_each { |item| any = true if yield item }
+    elsif class_passed
+      my_each { |item| any = true if item.is_a? class_passed }
+    else
+      my_each { |item| any = true if item }
+    end
+    any
+  end
+
+  def my_none?(class_passed = nil)
+    none = true
+    if block_given?
+      my_each { |item| none = false if yield item }
+    elsif class_passed
+      my_each { |item| none = false if item.is_a? class_passed }
+    else
+      my_each { |item| none = false if item }
+    end
+    none
+  end
+
+  def my_count(compare = nil)
+    count = 0
+    if block_given?
+      my_each { |item| count += 1 if yield item }
+    elsif !compare.nil?
+      my_each { |item| count += 1 if item == compare }
+    else
+      each { |_item| count += 1 }
+    end
+    count
+  end
+
+  def my_map(my_proc = nil)
+    map = []
+    if my_proc
+      my_proc.call
+    elsif block_given?
+      each { |item| map << yield(item) }
+    else
+      return to_enum
+    end
+    map
+  end
+
+  def my_inject(param = nil)
+    raise 'No block given' unless block_given?
+
+    memo = param || 0
+    my_each { |item| memo = yield(memo, item) }
+    memo
+  end
```

Comment:
#### :bug: 

- `my_inject` should be able to accept a `Symbol` as an operator (e.g `:+`, `:*`):
   ```ruby
   irb(main):055:0> [1,2,3].inject(:+)
   => 6
   irb(main):056:0> [1,2,3].my_inject(:+)
   RuntimeError (No block given)
   ```
- `my_inject` should be able to accept two arguments, first as an initial value and the other as a `Symbol`:

   ```ruby
   irb(main):057:0> [1,2,3].inject(5,:+)
   => 11
   irb(main):058:0> [1,2,3].my_inject(5,:+)
   RuntimeError (No block given)
   ```

### Find more details about [`inject?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject)